src/hypothesis/strategies.py

```diff
@@ -412,13 +413,17 @@ def sampled_from(elements):
 
     from hypothesis.searchstrategy.misc import SampledFromStrategy, \
         JustStrategy
-    elements = tuple(iter(elements))
```

Comment:
Or I just replace `tuple(iter(elements))` with `check_sample(elements)`, which makes the whole thing more consistent.  No length-one sets allowed!